core/corehttp/gateway_handler.go

```diff
@@ -63,8 +65,23 @@ func (i *gatewayHandler) loadTemplate() error {
 	return nil
 }
 
-func (i *gatewayHandler) ResolvePath(path string) (*dag.Node, error) {
-	return i.node.Resolver.ResolvePath(path)
+func (i *gatewayHandler) ResolvePath(p string) (*dag.Node, string, error) {
+	if strings.HasPrefix(p, "/ipns/") {
+		var err error
+		p, err = i.node.Namesys.Resolve(path.Clean(p[6:]))
```

Comment:
ipns can also point to a directory, so strings like: `/ipns/hash/a/b/c.avi` should be valid